LanguageExt.Core/Prelude_Union.cs

```diff
@@ -0,0 +1,373 @@
+﻿namespace LanguageExt
+{
+	using System;
+
+	public static partial class Prelude
+	{
+		public static Union<T1> Bind<A, T1>(this A a, Func<A, Union<T1>> func) => func(a);
+
+		public static Union<T1, T2> Bind<A, T1, T2>(this A a, Func<A, Union<T1, T2>> func) => func(a);
+
+		public static Union<T1, T2, T3> Bind<A, T1, T2, T3>(this A a, Func<A, Union<T1, T2, T3>> func) => func(a);
+
+		public static Union<T1, T2, T3, T4> Bind<A, T1, T2, T3, T4>(this A a, Func<A, Union<T1, T2, T3, T4>> func) => func(a);
+
+		public static Union<T1, T2, T3, T4, T5> Bind<A, T1, T2, T3, T4, T5>(this A a, Func<A, Union<T1, T2, T3, T4, T5>> func) => func(a);
+
+		public static Union<T1, T2, T3, T4, T5, T6> Bind<A, T1, T2, T3, T4, T5, T6>(this A a, Func<A, Union<T1, T2, T3, T4, T5, T6>> func) => func(a);
+
+		public static Union<T1, T2, T3, T4, T5, T6, T7> Bind<A, T1, T2, T3, T4, T5, T6, T7>(this A a, Func<A, Union<T1, T2, T3, T4, T5, T6, T7>> func) => func(a);
+
+		public static Union<T1, T2, T3, T4, T5, T6, T7, T8> Bind<A, T1, T2, T3, T4, T5, T6, T7, T8>(this A a, Func<A, Union<T1, T2, T3, T4, T5, T6, T7, T8>> func) => func(a);
+
+		public static Union<B> Select<A, B>(this Union<A> source, Func<Union<A>, B> predicate) =>
+			source.Bind(aval => new Union<B>(predicate(aval)));
+
+		//public static IUnion<TSource> Where<TSource>(this IUnion<TSource> source, Func<TSource, bool> predicate) =>
```

Comment:
If we don't know how to do this now, record the hoped-for interface elsewhere, but commented-out code is bad.